### ISSUE TITLE
[Broken Build]: Setting BaseOutputPath in Directory.Build.props Breaks Solution Loading

### ISSUE THREAD

**Issue by mbalous** [dotnet/msbuild#9445]
### Issue Description

I encountered an issue where adding a `Directory.Build.props` file with a custom `<BaseOutputPath>` breaks the loading of files and references in the entire solution. The goal is to set the output of all projects to the solution root.

### Steps to Reproduce

Create a `Directory.Build.props` file with the following content:
   ```xml
   <Project>
     <PropertyGroup>
       <BaseOutputPath>$(MSBuildThisFileDirectory)</BaseOutputPath>
     </PropertyGroup>
   </Project>
 ```

### Expected Behavior

The solution should load normally, with all projects outputting their build results to the solution's root directory.

### Actual Behavior

After adding the Directory.Build.props file, the solution fails to load correctly. No files are loaded, and no references are resolved.

msbuild doesn't see the `Program.cs` file.
```
PS D:\ConsoleApp10> tree /f
Folder PATH listing for volume Data
Volume serial number is 2221-5E74
D:.
│   ConsoleApp10.sln
│   Directory.Build.props
│
└───ConsoleApp10
        ConsoleApp10.csproj
        Program.cs

PS D:\ConsoleApp10> dotnet build
MSBuild version 17.7.3+4fca21998 for .NET
  Determining projects to restore...
  Restored D:\ConsoleApp10\ConsoleApp10\ConsoleApp10.csproj (in 62 ms).
CSC : error CS5001: Program does not contain a static 'Main' method suitable for an entry point [D:\ConsoleApp10\ConsoleApp10\ConsoleApp10.csproj]

Build FAILED.

CSC : error CS5001: Program does not contain a static 'Main' method suitable for an entry point [D:\ConsoleApp10\ConsoleApp10\ConsoleApp10.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.04
```

**Comment by AR-May** [comment]
Team triage: @mbalous This behavior does not sound right. You might be interested though in this dotnet feature: https://learn.microsoft.com/en-us/dotnet/core/sdk/artifacts-output

**Comment by JaynieBai** [comment]
@AR-May  I repro this issue also.  When add <BaseOutputPath>$(MSBuildThisFileDirectory)</BaseOutputPath> in the .NET project, it doesn't add the Compile Item when evaluate the project. So the error CS5001: Program does not contain a static 'Main' method suitable for an entry point [D:\ConsoleApp10\ConsoleApp10\ConsoleApp10.csproj] happens.

**Comment by rainersigwald** [comment]
> if $(BaseOutputPath) is the solution root, then none of the files in that tree are added as items by default.
> 
> I think it's a niche case and a solution that sets BaseOutputPath in this way should configure its own DefaultItemExcludes, or not use default items.

That makes sense to me and I agree: we shouldn't add logic to handle this special case. Thanks for the analysis!

**Comment by mbalous** [comment]
> "no references are resolved" seems unexpected though. Does this mean ProjectReference?

yes that's what i mean. The projects in Visual Studio seems completely empty. No references to other projects, no source files loaded.

**Comment by mbalous** [comment]
> I think it's a niche case and a solution that sets BaseOutputPath in this way should configure its own DefaultItemExcludes, or not use default items.

I think it's not that unusual to set BaseOutputPath to the solution root. Can you advise me the way how to achieve that? 
Thank you, @KalleOlaviNiemitalo !
